js/content-slider.js

```diff
@@ -1,85 +1,81 @@
 
-      //get the DOMRect objects of the emelements to use in positioning calculations
-      let contentBlockUpperRect     = contentBlockUpper.getBoundingClientRect();
-      let contentSliderControlsRect = contentSliderControls.getBoundingClientRect();
+      let intitialTopValue = this.contentSliderControlsRect.top; //79px
 
-      //set height of Left and Right controls
-      //height must be set on individual controls as they are empty
-      controlsLeft.style.height     = `${contentBlockUpperRect.height}px`;
-      controlsRight.style.height    = `${contentBlockUpperRect.height}px`;
+      this.contentSliderControls.style.top = `${this.controlSurfaceRect.top - intitialTopValue}px`;
 
-      // console.log("initial contentBlockUpperRect.top",contentBlockUpperRect.top);
-      // console.log("initial contentSliderControlsRect.top", contentSliderControlsRect.top);
-
-      // viewport starts at [0, 79px] ???
-      let intitialTopValue = contentSliderControlsRect.top; //79px
-      let diff = 0;
-
-      contentSliderControls.style.top = `${contentBlockUpperRect.top - intitialTopValue}px`;
-      
-      let resizeObserver = new ResizeObserver(entries => {
-        for (let entry of entries){
-        //   recalculate the controls rect.
-          contentSliderControlsRect = contentSliderControls.getBoundingClientRect();
+    } else {
+      console.error(
+        "[Content Slider] No content slider found. Please make sure the .className parameter passed to the content slider exists and is correct."
+      );
+    }
+  }
 
-          //find the diff between controlRect and upper rect 
-          diff = contentBlockUpperRect.top - contentSliderControlsRect.top;
+  resizeControlSurface() {
+    this.controlSurfaceRect = this.controlSurface.getBoundingClientRect();
+    this.contentSliderControlsRect = this.contentSliderControls.getBoundingClientRect();
 
-        //  Adjust controls' top - math is incorect
-          // contentSliderControls.style.top = `${contentBlockUpperRect.top - intitialTopValue + diff}px`;
-        }
-      });
-      
-      resizeObserver.observe(this.contentSlider);
+    this.controlsLeft.style.height = `${this.controlSurfaceRect.height}px`;
+    this.controlsRight.style.height = `${this.controlSurfaceRect.height}px`;
+  }
 
-      //event listeners
-      controlsLeft.addEventListener("click", (e) => {
-        e.preventDefault();
-        contentSliderCardsList.scrollLeft -= 500;
-        
-      })
+  initializeControlSurface() {
+    this.contentSliderCardsList = this.contentSlider.querySelector(
+      ".content-slider__cards-list"
+    );
+    this.contentSliderControls = this.contentSlider.querySelector(
+      ".content-slider__controls"
+    );
+    this.controlsLeft = this.contentSliderControls.querySelector(
+      ".content-slider__controls-left"
+    );
+    this.controlsRight = this.contentSliderControls.querySelector(
+      ".content-slider__controls-right"
+    );
+  }
```

Comment:
These feels like needless abstraction. Is there a reason to not just do this in the constructor?